std/internal/cstring.d

```diff
@@ -132,55 +132,69 @@ auto tempCString(To = char, From)(From str)
         }
 
         To[256 / To.sizeof] _buff;  // the 'small string optimization'
+
+        static Res trustedVoidInit() { Res res = void; return res; }
     }
 
-    Res res = void;     // expensive to fill _buff[]
+    Res res = Res.trustedVoidInit();     // expensive to fill _buff[]
 
     // Note: res._ptr can't point to res._buff as structs are movable.
 
-    import std.utf : byUTF;
-
-    To* p      = res._buff.ptr;
-    size_t len = res.buffLength;
+    To[] p = res._buff[0 .. Res.buffLength];
     size_t i;
 
+    static To[] trustedRealloc(To[] buf, size_t i, To* resptr, size_t strLength)
+        @trusted @nogc nothrow
```

Comment:
Given how trustedRealloc may invalidate buf, I think the better interface is to take it as a ref, to avoid issues where the caller doesn't re-assign it. Otherwise, you are left with a dangling pointer. This doesn't prevent all mistakes, but it does support the common case.